pyaedt/modules/solutions.py

```diff
@@ -1438,492 +1352,325 @@ def assign_weight(self, a, b, taper="flat", port_cont=0):
+            return False
+        data_to_plot = np.reshape(data_to_plot, (data["nTheta"], data["nPhi"]))
+        th, ph = np.meshgrid(data["Theta"], data["Phi"])
+        if show:
+            return plot_contour(
+                x=th,
+                y=ph,
+                qty_to_plot=data_to_plot,
+                xlabel="Theta (degree)",
+                ylabel="Phi (degree)",
+                title=title,
+                levels=levels,
+                snapshot_path=export_image_path,
+            )
+        else:
+            return data_to_plot
+
+    # fmt: off
     @pyaedt_function_handler()
     def plot_2d_cut(
-        self,
-        qty_str="RealizedGain",
-        primary_sweep="phi",
-        secondary_sweep_value=0,
-        phi_scan=0,
-        theta_scan=0,
-        title="Far Field Cut",
-        convert_to_db=True,
-        export_image_path=None,
+            self,
+            farfield_quantity="RealizedGain",
+            primary_sweep="phi",
+            secondary_sweep_value=0,
+            phi_scan=0,
+            theta_scan=0,
+            title="Far Field Cut",
+            format_quantity="dB10",
+            export_image_path=None,
+            show=True,
+            is_polar=False,
+            **kwargs
     ):
+        # fmt: on
         """Create a 2D plot of specified quantity in matplotlib.
 
         Parameters
         ----------
-        qty_str : str, optional
+        farfield_quantity : str, optional
             Quantity to plot. Default `"RealizedGain"`.
+            Available functions are: `"RealizedGain"`, `"RealizedGain_Theta"`, "`RealizedGain_Phi"`, `"rETotal"`,
+            `"rETheta"`, `"rEPhi"`.
         primary_sweep : str, optional.
             X Axis variable. Default is `"phi"`. Option is  `"theta"`.
         secondary_sweep_value : float, list, string, optional
             List of cuts on secondary sweep to plot. Options are `"all"`, single value float or list of float.
         phi_scan : float, int, optional
-            Phi Scan Angle in degree. Default `0`.
+            Phi scan angle in degree. Default `0`.
         theta_scan : float, int, optional
-            Theta Scan Angle in degree. Default `0`.
+            Theta scan angle in degree. Default `0`.
         title : str, optional
             Plot title. Default `"RectangularPlot"`.
-        convert_to_db : bool, optional
-            Either if the quantity has to be converted in db or not. Default is `True`.
+        format_quantity : str, optional
+            Conversion data function.
+            Available functions are: `"dB10"`, `"dB20"`, "`abs"`, `"real"`, `"imag"`, `"norm"`, `"ang"`, `"and_deg"`.
         export_image_path : str, optional
             Full path to image file. Default is None to not export.
-
+        show : bool, optional
+            Whether to show the plot or return the matplotlib object. Default is `True`.
+        is_polar : bool, optional
+            Set to `True` if this is a polar plot.
```

Comment:
```suggestion
            Whether this plot is a polar plot. The default is ``True``.
```
What kind of plot is it if this paramter is set to ``False``. I think we need to add something like this: "If ``False``, the plot is an X plot."